### PR TITLE
Keep UserAgentName style consistent

### PR DESCRIPTION
**Pull request by NickrenREN**
Keep using UserAgentName for controllers  and add some logs for debugging

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: federation/pkg/federation-controller/ingress/ingress_controller.go [hunk 1]
```diff
@@ -63,6 +63,7 @@ const (
 	// other cluster.
 	firstClusterAnnotation = "ingress.federation.kubernetes.io/first-cluster"
 	ControllerName         = "ingresses"
+	UserAgentName          = "Federation-Ingresses-Controller"
```
Comment (nikhiljindal) [inline federation/pkg/federation-controller/ingress/ingress_controller.go]: I see sync controller uses user agent of the form "ingress-controller". We should definitely add "federation" there.
Please update https://github.com/NickrenREN/kubernetes/blob/8862669061aa4e5aaf8284da746ad6f69f765133/federation/pkg/federation-controller/sync/controller.go#L91.
cc @marun @perotinus

File: federation/pkg/federation-controller/replicaset/replicasetcontroller.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ import (
 const (
 	FedReplicaSetPreferencesAnnotation = "federation.kubernetes.io/replica-set-preferences"
 	allClustersKey                     = "THE_ALL_CLUSTER_KEY"
-	UserAgentName                      = "Federation-replicaset-Controller"
+	UserAgentName                      = "Federation-Replicaset-Controller"
 	ControllerName                     = "replicasets"
```
Comment (nikhiljindal) [inline federation/pkg/federation-controller/replicaset/replicasetcontroller.go]: Not in this PR, but I see a potential to merge ControllerName and UserAgentName

File: federation/cmd/federation-controller-manager/app/options/options.go [hunk 3]
```diff
@@ -115,6 +120,7 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.ServiceDnsSuffix, "service-dns-suffix", s.ServiceDnsSuffix, "DNS Suffix to use when publishing federated service names.  Defaults to zone-name")
 	fs.IntVar(&s.ConcurrentServiceSyncs, "concurrent-service-syncs", s.ConcurrentServiceSyncs, "The number of service syncing operations that will be done concurrently. Larger number = faster endpoint updating, but more CPU (and network) load")
 	fs.IntVar(&s.ConcurrentReplicaSetSyncs, "concurrent-replicaset-syncs", s.ConcurrentReplicaSetSyncs, "The number of ReplicaSets syncing operations that will be done concurrently. Larger number = faster endpoint updating, but more CPU (and network) load")
+	fs.IntVar(&s.ConcurrentDeploymentSyncs, "concurrent-deployment-syncs", s.ConcurrentDeploymentSyncs, "The number of Deployments syncing operations that will be done concurrently. Larger number = faster endpoint updating, but more CPU (and network) load")
```
Comment (NickrenREN) [inline federation/cmd/federation-controller-manager/app/options/options.go]: But here we did not share ConcurrentReplicaSetSyncs
https://github.com/kubernetes/kubernetes/blob/fc6eb52d0503c8353596a0b8d2b393fe7ed0041c/cmd/kube-controller-manager/app/options/options.go#L131

**Comment by NickrenREN** [comment]
Hope I have addressed all your comments @nikhiljindal

**Comment by nikhiljindal** [comment]
LGTM.

Will add lgtm, once you have verified that presubmit federation e2e is passing on this PR.

Why did you revert ConcurrentDeploymentSyncs changes? If you plan to send a different PR for it, then please update the title for this PR.

**Comment by nikhiljindal** [comment]
Also to clarify my [comment](https://github.com/kubernetes/kubernetes/pull/45218#discussion_r114451999) for sharing flags with kube-controller-manager, I wasnt suggesting sharing ConcurrentDeploymentSyncs and ConcurrentReplicaSetSyncs, but I was suggesting sharing both those flags in kube-controller-manager and federation-controller-manager. Right now, we have code duplication at both places. We need to refactor that code.  Ref https://github.com/kubernetes/kubernetes/issues/24996

**Comment by NickrenREN** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by NickrenREN** [comment]
I will send another PR for ConcurrentDeploymentSyncs

**Comment by NickrenREN** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by NickrenREN** [comment]
@nikhiljindal presubmit federation e2e passed
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/45218/pull-kubernetes-federation-e2e-gce/3185/

File: federation/pkg/federation-controller/sync/controller.go [hunk 4]
```diff
@@ -88,7 +88,7 @@ type FederationSyncController struct {
 
 // StartFederationSyncController starts a new sync controller for a type adapter
 func StartFederationSyncController(kind string, adapterFactory federatedtypes.AdapterFactory, config *restclient.Config, stopChan <-chan struct{}, minimizeLatency bool) {
-	restclient.AddUserAgent(config, fmt.Sprintf("%s-controller", kind))
+	restclient.AddUserAgent(config, fmt.Sprintf("federation-%s-controller", kind))
```
Comment (nikhiljindal) [inline federation/pkg/federation-controller/sync/controller.go]: Sorry for not noticing that earlier. You need to change line 106 as well

**Comment by nikhiljindal** [comment]
Great. Thanks for the fixes. Sorry found one last comment before merging.

File: federation/pkg/federation-controller/sync/controller.go (hunk 4 above)
Comment (NickrenREN) [inline federation/pkg/federation-controller/sync/controller.go]: Thanks, done

**Pull request merged**